### DISCUSSION TITLE
WebSocket don't send heartbeat after reconnection

### DISCUSSION THREAD

**Discussion by Harokku** [solidjs-community/solid-primitives#527] [Q&A] [answered]
Hi,
1st of all thanks for the great job, definitely love it.

I've noticed an issue with WebSocket makeHearthbeatWS method:

At first connection everything work fine: ping are sent, pong are received and the client try to reconnect on connection lost

After reconnection ping messages stop from being send to the server, neither pong are received (checked dev console, server console and even with "evil" msg.log on event listener)
In this state if the server close the connection the client try to reconnect (and successfully do it) but still no heartbeat are sent.

Everything restart working fine after any message is received (even if sent from another client)

It's intended to work like this or I'm missing something in my code?
(In my simple test server it don't send welcome msg after a client connect, I think in a "real life" scenario this issue could be completely negligible)

Next the code I've used:

```js
import {   
    createReconnectingWS,
    createWSState,
    makeHeartbeatWS
} from "@solid-primitives/websocket";

const socket = makeHeartbeatWS(
        createReconnectingWS(`ws://localhost:3000/v1/ws/chat`, undefined, {delay: 10000}),
        {message: "ping", interval: 5000, wait: 7500},
    );
```

**Comment by atk** [comment, marked as answer]
I found the error. I only do the setup once, but do not repeat it on reconnect. Thank you for your issue, I will prepare a fix within the next days.

**Reply by atk** [reply]
I have created an issue from this discussion and am already preparing a fix.
